pkg/controller/nodeipam/ipam/controller_test.go

```diff
@@ -1,3 +1,6 @@
+//go:build !providerless
+// +build !providerless
```

Comment:
why is this providerless? I have to check this because there are several controllers, but only one is provider specific

Reply:
this was depending on a method that was not available, added a new commit at the end of the chain now to move that method to a common spot.

https://github.com/kubernetes/kubernetes/pull/122595/commits/b7012da2ad4deb7d3e8fbb24a19341646ad5a1cd

Reply:
spotted this too https://github.com/kubernetes/kubernetes/pull/122645